### ISSUE TITLE
Web-to-Lead: Double-Oprt-In: Confirmation link causes Error 500

### ISSUE THREAD

**Issue by agri17** [espocrm/espocrm#2056]
[EspoCRM Version 6.1.4, PHP 7.2.24-0ubuntu0.18.04.8, mysql  Ver 14.14 Distrib 5.7.35, Apache/2.4.29; FF 91.0.1 (64-Bit)]

I try to capture a lead from an online form as described in https://docs.espocrm.com/administration/web-to-lead/, with just some more data I've put into hidden input fields. This works well, if I do *not* use Double-Opt-In.
Once I activate Double-Opt-In, when a user fills in the form and sends it, they get a confirmation mail. Data is not stored jet in Espo. BUT: when user clicks the confirmation link, they get 
`500 Internal Server Error
ArrayValue: Bad value passed to JSON_ARRAY attribute sektorentyp.`
Still the lead is set up, it only lacks the target list. If out of curiosity, the confirmation link is clicked a second time, there is a 404 Error (obviously), but what's stange is, those lost data is filled in.

Part of logfile (/data/logs/...):
[espo-2021-08-19(part).log](https://github.com/espocrm/espocrm/files/7014675/espo-2021-08-19.part.log)
My html-form:
[radlform.zip](https://github.com/espocrm/espocrm/files/7014686/radlform.zip)

**Comment by yurikuzn** [comment]
Maybe you have `sektorentyp` attribute set in some different place with not proper value (non-array value).

**Comment by agri17** [comment]
'sektorentyp' is Multi-Enum.
If I set up a new lead from the backend and even leave sektorentyp untouched, all works well, no error.
If I set up a lead via lead-capture (Html form. The form does not use sektorentyp. Should be untouched, then.) without Double-opt-in, all works well.
If I use Double-opt-in (same html form) and click the confirmation link, I get error 500.
I cannot see, where false data for sektorentyp should come from. To me it looks very much to be a problem of double-opt-in.

**Comment by yurikuzn** [comment]
Could you help with debugging? https://github.com/espocrm/espocrm/blob/6.1.8/application/Espo/Tools/LeadCapture/LeadCapture.php#L357

**Comment by agri17** [comment]
First of all, I must appologize! 
When I first used LeadCapture without Double-Opt-In, I did _not get any error page_ concerning an error 500. I've now realized per developper tools, that there **is** the error even without DOI. 
To get rid of the " Bad value passed to JSON_ARRAY attribute sektorentyp", I added another hidden field to my form such as
<input type="hidden" name="sektorentyp" value='["Wissenschaft und Bildung"]'>
and adjusted js and LeadCapture accordingly. "Wissenschaft und Bildung" will the be correctly insertet into attribute sectorentyp. Still I get the same error, complaining about "Bad value" passed to attribute sektorentyp. 
There it no problem - no error! - if I send the form a second time, so that's to say, when the lead already exists.
Concerning debugging the code: I'm not too familiar with php, I must confess ...
